tests/commands/test-kob-install.sh

```diff
@@ -1 +1,328 @@
+  rm -rf ~/Dev_$kobman_env_name
+  rm -rf $path_to_kob_envs/kob_env_$kobman_env_name
+  sed -i "s/dummyenv,0.0.2,0.0.3,0.0.5,0.0.7,0.0.9//g" $KOBMAN_DIR/var/list.txt 
+
+}
+
+
+
+
+function __test_kob_install_dummyenv_script(){
+  
+cat <<EOF
+#!/bin/bash
+
+
+function __kobman_install_dummyenv
+{
+  
+  cd ~
+  if [ ! -d "Dev_dummyenv" ]; then 
+    
+    __kobman_create_dev_environment 
+    __kobman_echo_white "dummyenv installed"
+  else
+    
+    rm -rf ~/Dev_dummyenv	
+    __kobman_create_dev_environment  
+    __kobman_echo_white "dummyenv installed"
+    
+  fi		
+}
+function __kobman_uninstall_dummyenv
+{
+    __kobman_echo_white "uninstall"
+}
+function __kobman_update_dummyenv
+{
+    __kobman_echo_white "update"
+}
+function __kobman_upgrade_dummyenv
+{
+    __kobman_echo_white "upgraded"
+
+}
+function __kobman_start_dummyenv
+{
+    __kobman_echo_white "start"
+}
+function __kobman_stop_dummyenv
+{
+    __kobman_echo_white "stop"
+}
+function __kobman_create_dev_environment 
+{
+	 
+	cd ~
+	mkdir -p Dev_dummyenv
+	cd Dev_dummyenv
+	export DUMMYENV_DEV_DIR=$PWD
+	mkdir -p test/ dependency/
+}
+
+EOF
+}
+
+function __test_kob_dummyenv_validation()
+{
+
+cat <<EOF
+  #!/bin/bash
+  status="true"
+  function __test_kob_dummyenv_run
+  {
+    __test_kob_init || return 1
+    
+    __test_kob_execute
+    
+    __test_kob_validate || return 1
+    __test_kob_dummyenv_cleanup
+    __test_kob_output
+  }
+  function __test_kob_init
+  {
+    
+    if [[ ! -d ~/Dev_dummyenv ]]; then
+      
+      status="false"
+      __test_kob_output
+      return 1
+    else
+      
+      return 0
+
+    fi
+
+  }
+  function __test_kob_execute
+  {
+    
+    touch ~/dummy.txt
+    # echo "output of execute of dummyenv" > ~/dummy.txt
+    
+
+    
+
+  }
+  function __test_kob_validate
+  {
+    
+    
+    ls ~ | grep -qw "dummy.txt"
+    if [[ "$?" != "0" ]]; then
+      status="false"
+      __test_kob_output
+      __test_kob_dummyenvcleanup
+      return 1
+    fi
+
+    if [[ "$(cat $KOBMAN_DIR/var/kobman_env_dummyenv.proc)" == "1" ]]; then
+      
+      status="false"
+      __test_kob_output
+      __test_kob_dummyenvcleanup
+      return 1
+
+    fi
+
+    commands="uninstall update upgrade start stop"
+    for i in $commands; do
+        __kobman_$i_dummyenv | grep $i
+        if [[ "$?" != "0" ]]; then
+            __kobman_echo_white "error with $i"
+            status="false"
+            __test_kob_dummyenvcleanup
+            __test_kob_output
+            return 1
+        fi
+    done
+    status="true"
+    
+    __test_kob_output
+    return 0 
+  }
+
+  function __test_kob_dummyenv_cleanup
+  {
+    rm -rf ~/dummy.txt
+  }
+
+  function __test_kob_output
+  {
+    
+    if [[ $status == "true" ]]; then
+      echo "success"
+    else
+      echo "failed"
+    fi
+  }
+  __test_kob_dummyenv_run
+
+
+EOF
+}
+
+function __test_kob_insert_dummyenv_version
```

Comment:
Rename as fake_ publish_dummyenv as it is seems to be a helper function not a function to  kob_insert_dummyenv_version

Please add a TODO to make use of __kobman_secure_curl instead of curl after __kobman_utils.sh is refactored out of command folder.

Reply:
I am sorry Vinod, can you please explain what you said last